python/pyarrow/_parquet_encryption.pyx

```diff
@@ -473,9 +529,167 @@ cdef class CryptoFactory(_Weakrefable):
     def remove_cache_entries_for_all_tokens(self):
         self.factory.get().RemoveCacheEntriesForAllTokens()
 
+    def rotate_master_keys(
+            self,
+            KmsConnectionConfig kms_connection_config,
+            parquet_file_path, 
+            FileSystem filesystem,
+            double_wrapping = True,
+            cache_lifetime_seconds = 600):
+        """ Rotates master encryption keys for a Parquet file that uses
+        external key material.
+
+        Parameters
+        ----------
+        kms_connection_config : KmsConnectionConfig
+            Configuration of connection to KMS
+
+        parquet_file_path : str or pathlib.Path
+            Path to a parquet file using external key material.
+
+        filesystem : FileSystem, default None
+            Used only when the parquet file uses external key material. If
+            nothinng passed, will be inferred based on parquet_file_path. 
+
+        double_wrapping : bool, default True
+            In the single wrapping mode, decrypts data keys with old master
+            keys, then encrypts them with new master keys. In the double
+            wrapping mode, decrypts KEKs (key encryption keys) with old 
+            master keys, generates new KEKs and encrypts them with new master
+            keys.
```

Comment:
This comment sounds like it's assuming that the `double_wrapping` parameter matches the existing state of the file, but I think it's possible to use this method to convert from single to double wrapping or vice-versa. How about something like:
```suggestion
            In the single wrapping mode, encrypts data encryption keys with
            new master keys. In the double wrapping mode, generates new
            KEKs (key encryption keys) and uses these to encrypt the data keys,
            and encrypts the KEKs with the new master keys.
```